okhttp/src/main/java/okhttp3/internal/connection/Exchange.java

```diff
@@ -215,17 +215,15 @@ public void noRequestBody() {
         super.write(source, byteCount);
         this.bytesReceived += byteCount;
       } catch (IOException e) {
-        complete(e);
-        throw e;
+        throw complete(e);
```

Comment:
This doesn't read quite right.  I can see what it's doing, and it looks correct, just feels a bit off.

Even reading into exchangeMessageDone there is a lot of assumptions about the nullability of e

Reply:
One invariant we do hold: if the parameter in is non-null, the return value is non-null.

Still, if you’d like to take a pass at tidying this up please do!